src/containers/List/ListFilters/ListFilters.tsx

```diff
@@ -0,0 +1,145 @@
+  string.split(',').filter((option) => !!option.trim())
+
+const getDisplayableFilters = (filterDefinitions: FilterDefinitions) => {
+  const result: FilterDefinitions = {}
+  Object.entries(filterDefinitions).forEach(([filterName, filterValue]) => {
+    // Missing title determines if filters should be shown in UI or not
+    if (!filterValue.title) return
+    result[filterName] = filterValue
+  })
+
+  return result
+}
+
+const ListFilters: React.FC<{ filterDefinitions: FilterDefinitions; filterListParameters: any }> =
+  ({ filterDefinitions, filterListParameters }) => {
+    const { query, updateQuery } = useRouter()
+
+    const displayableFilters = getDisplayableFilters(filterDefinitions)
+
+    const filterNames = Object.keys(displayableFilters)
+    const [activeFilters, setActiveFilters] = useState<string[]>([])
+
+    // Add fitlers from URL to activeFilters, filters may not have criteria/options yet thus we have to use activeFilter state variable
+    useEffect(() => {
+      const filersInQuery = filterNames.filter((filterName) => !!query[filterName])
+      const newFilters = filersInQuery.filter((filterName) => !activeFilters.includes(filterName))
+      setActiveFilters([...activeFilters, ...newFilters])
+    }, [query])
+
+    // Filter criteria/options states is provided by query URL, methods below are get and set query fiter criteria
+    const getMethodsForOptionFilter: GetMethodsForOptionFilter = (filterName) => ({
+      getActiveOptions: () => getArrayFromString(query[filterName]),
+      setActiveOption: (option: string) =>
+        updateQuery({ [filterName]: [...getArrayFromString(query[filterName]), option].join(',') }),
+      setInactiveOption: (option: string) =>
+        updateQuery({
+          [filterName]: getArrayFromString(query[filterName])
+            .filter((_option) => _option != option)
+            .join(','),
+        }),
+    })
+
+    const addFilter = (filterName: string) => setActiveFilters([...activeFilters, filterName])
+    const getOnRemove = (filterName: string) => () => {
+      setActiveFilters(activeFilters.filter((_filterName) => _filterName !== filterName))
+      updateQuery({
+        [filterName]: undefined,
+      })
+    }
+
+    const availableFilterNames = filterNames.filter(
+      (filterName) => !activeFilters.includes(filterName)
+    )
+
+    return (
+      <div id="list-filter">
+        <Dropdown text="+ Filter" icon={null}>
+          <Dropdown.Menu>
+            {availableFilterNames.map((filterName) => (
+              <Dropdown.Item key={filterName} onClick={() => addFilter(filterName)}>
+                {displayableFilters[filterName].title}
+              </Dropdown.Item>
+            ))}
+          </Dropdown.Menu>
+        </Dropdown>
+
+        {activeFilters.map((filterName) => {
+          const filter = displayableFilters[filterName]
+          if (!filter.title) return null
+
+          if (filter.type === 'enumList') {
```

Comment:
I recommend using switch case here - it's just easier to follow although you need to have a default case, I would use the enumList which seems to be the most used one.

The code is looking great though, congrats!

Reply:
fa1860d3aca2cd4f8b3bc2473e1220ef920021c8
so much better, thanks